File: solutions/cpp/sieve/1/sieve.cpp
#include "sieve.h"

namespace sieve {

    std::vector<int> primes(int upper_limit){
        std::bitset<MAX> prime_mask;
        std::vector<int> primes;
        for(int i = 2; i <= upper_limit; ++i){
            if(!prime_mask[i]){
                primes.push_back(i);
                for(int j = i * i; j <= upper_limit ; j += i){
                    prime_mask[j] = true;
                }
            }
        }
        return primes;
    }

}  // namespace sieve

File: solutions/csharp/sieve/1/Sieve.cs
public static class Sieve
{
    public static int[] Primes(int limit)
    {
        bool[] sieve = new bool[limit + 1];
        List<int> primes = new();
        for(int i = 2; i <= limit; i++){
            if(!sieve[i]){
                primes.Add(i);
                for(int j = i + i; j <= limit; j+=i){
                    sieve[j] = true;
                }
            }
        }
        return primes.ToArray();
    }
}